src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -7921,6 +7923,21 @@ public static void Main()
                 Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "lam").WithArguments("arg2", "<anonymous delegate>").WithLocation(6, 9));
         }
 
+        [Fact]
+        public void ParamsArray_Langversion()
+        {
+            var source = """
+                var lam = (params int[] xs) => xs.Length;
+                """;
+
+            CreateCompilation(source, parseOptions: TestOptions.Regular11).VerifyDiagnostics(
+                // (1,12): error CS8652: The feature 'lambda params array' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // var lam = (params int[] xs) => xs.Length;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "params").WithArguments("lambda params array").WithLocation(1, 12));
+
+            CreateCompilation(source).VerifyDiagnostics();
```

Comment:
We want to use `TestOptions.RegularNext` here, so that when C# 12 is released we make sure we're testing the boundary condition correctly.

Reply:
Right, thanks. Fixed: 2fdeb13d8652f5a2b2e13cb88bcf222e51ec7a16